File: trunk/RRComSSys/Projects/CVM/RRComSSys.CVM/src/edu/fiu/cis/rrcomssys/cvm/mte/xcmlmodel/Connection.java
package edu.fiu.cis.rrcomssys.cvm.mte.xcmlmodel;

import java.util.LinkedList;
import java.util.List;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.ElementList;

public class Connection
{
	/* Attributes */
	@Attribute(name = "connectionID", required = true)
	private String _id = "";

	@Attribute(name = "bandwidth", required = false)
	private String _bandwidth = "";

	/* Elements */
	@ElementList(entry = "device", inline = true, required = false)
	private List<Device> _deviceList = new LinkedList<Device>();

	@ElementList(entry = "mediumTypeNameRef", inline = true, required = false)
	private List<String> _mediaRefList = new LinkedList<String>();

//	@ElementList(entry = "formTypeNameRef", inline = true, required = false)
//	private List<String> _formRefList = new LinkedList<String>();

	/* Member Variables */
	private List<Medium> _media = new LinkedList<Medium>();

	/* Constructors & Initializers */
	public Connection()
	{}

	/* Properties */
	public String getID()
	{
		return _id;
	}

	public void setID(String value)
	{
		_id = value;
	}

	public String getBandwidth()
	{
		return _bandwidth;
	}

	public void setBandwidth(String value)
	{
		_bandwidth = value;
	}

	public List<Device> getDevices()
	{
		return _deviceList;
	}

	public List<Medium> getMedia()
	{
		return _media;
	}

	/* Private Methods */
	void initializeMedia(List<Medium> media)
	{
		for (String mediaRef : _mediaRefList)
		{
			Medium medium = getMediumByName(media, mediaRef);
			if(medium==null)
				continue;
			_media.add(medium);
		}
	}

	private Medium getMediumByName(List<Medium> mediumList, String name)
	{
		for (Medium medium : mediumList)
			if (medium.getName().equals(name))
				return medium;
		return null;
	}
}

File: trunk/RRComSSys/Projects/CVM/RRComSSys.CVM.ObjectModel/XCMLModel/Connection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;

namespace RRComSSys.CVM.ObjectModel.XCMLModel
{
	public class Connection
	{
		#region Member Variables
		private String _id = "";
		private String _bandwidth = "";
		private List<Device> _deviceList = new List<Device>();
		private List<String> _mediaRefList = new List<String>();
		private List<Medium> _media = new List<Medium>();
		#endregion

		#region Properties
		[XmlAttribute(AttributeName = "connectionID")]
		public String ID
		{
			get { return _id; }
			set { _id = value; }
		}

		[XmlAttribute(AttributeName = "bandwidth")]
		public String Bandwidth
		{
			get { return _bandwidth; }
			set { _bandwidth = value; }
		}

		[XmlElement(ElementName = "device")]
		public List<Device> Devices
		{
			get { return _deviceList; }
			set { _deviceList = value; }
		}

		[XmlElement(ElementName = "mediumTypeNameRef")]
		public List<String> MediaNameReferences
		{
			get { return _mediaRefList; }
			set { _mediaRefList = value; }
		}

		[XmlIgnore]
		public List<Medium> Media
		{
			get { return _media; }

		}
		#endregion

		#region Public Methods
		public Medium GetMediumByName(List<Medium> mediumList, String name)
		{
			foreach (Medium medium in mediumList)
				if (medium.Name.Equals(name))
					return medium;
			return null;
		}
		#endregion

		#region Private Methods
		internal void InitializeMedia(List<Medium> media)
		{
			foreach (String mediaRef in _mediaRefList)
			{
				Medium medium = GetMediumByName(media, mediaRef);
				if (medium == null)
					continue;
				_media.Add(medium);
			}
		}
		#endregion
	}
}
